src/types/timestamp.rs

```diff
@@ -572,6 +572,66 @@ impl Timestamp {
     pub fn precision(&self) -> Precision {
         self.precision
     }
+
+    /// Returns the year that has been specified in the [Timestamp].
+    pub fn year(&self) -> i32 {
+        self.date_time.year()
+    }
+
+    /// Returns the month that has been specified in the [Timestamp].
+    pub fn month(&self) -> u32 {
+        self.date_time.month()
+    }
+
+    /// Returns the day that has been specified in the [Timestamp].
+    pub fn day(&self) -> u32 {
+        self.date_time.day()
+    }
+
+    /// Returns the hour(s) that has been specified in the [Timestamp].
+    pub fn hour(&self) -> u32 {
+        // verify if the timestamp has an offset
+        if let Some(offset) = self.offset {
+            // `NativeDateTime#hours()` returns hours normalized as per UTC
+            // for local time we need to +/- the difference
+            return (self.date_time.hour() as i32 + (offset.local_minus_utc() / 3600)) as u32;
+        }
+        self.date_time.hour()
+    }
+
+    /// Returns the minute(s) that has been specified in the [Timestamp].
+    pub fn minute(&self) -> u32 {
+        // verify if the timestamp has an offset
+        if let Some(offset) = self.offset {
+            // `NativeDateTime#hours()` returns minutes normalized as per UTC
+            // for local time we need to +/- the difference
+            let local_minus_utc_minute = (offset.local_minus_utc() % 3600) / 60;
+            return (self.date_time.minute() as i32 + local_minus_utc_minute) as u32;
+        }
+        self.date_time.minute()
+    }
+
+    /// Returns the second(s) that has been specified in the [Timestamp].
+    pub fn second(&self) -> u32 {
+        self.date_time.second()
+    }
+
+    /// Returns the UTC hour(s) that has been specified in the [Timestamp].
+    pub fn utc_hour(&self) -> u32 {
```

Comment:
Idea—we could get rid of the `xyz()`/`utc_xyz()` dichotomy entirely by making all of the functions return the value in the "local"/offset time, and then adding one function to convert to a UTC timestamp.

I'm not quite sure myself whether this is a good idea, which is why I've labelled it an "idea" rather than a "suggestion", but I hope that by presenting the idea, others might chime in.

Reply:
It's probably pretty cheap to do this. Users would likely appreciate having a `fn to_utc(&self) -> Timestamp` method anyway.

Reply:
Reference issue: https://github.com/amazon-ion/ion-rust/issues/489